ValetTests/ValetTests.m

```diff
@@ -90,15 +91,20 @@ - (void)setUp;
     self.string = @"bar";
     self.secondaryString = @"bar2";
     self.additionalValets = [NSMutableArray new];
+    self.manualKeychainEntries = [NSMutableArray<NSDictionary *> new];
 }
 
 - (void)tearDown;
 {
-    [super tearDown];
-    
     for (VALValet *const additionalValet in self.additionalValets) {
         [additionalValet removeAllObjects];
     }
+
+    for (NSDictionary *keychainEntry in self.manualKeychainEntries) {
```

Comment:
ah, what about a finally block? I didn't know this existed and it does the same thing i wanted to get out of `tearDown()`: something that is ensured to run regardless of whether an exception is thrown and execution stops.

```objective-c
NSDictionary *keychainEntry = <..>

@try {
.. all the code
}
@finally {
  SecItemDelete(keychainEntry);
}
```